src/core/manager-serialize.c

```diff
@@ -513,10 +519,21 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                 else if ((val = startswith(l, "exec-runtime=")))
                         (void) exec_runtime_deserialize_one(m, val, fds);
                 else if ((val = startswith(l, "subscribed="))) {
-
                         if (strv_extend(&m->deserialized_subscribed, val) < 0)
                                 return -ENOMEM;
+                } else if ((val = startswith(l, "vl-ss-address="))) {
```

Comment:
Hmm, if we already have a varlink server in `m->varlink_server`, shouldn't we skip deserialization here? In other words, I think we should not create a varlink server before deserialization, and here we should assert that `m->varlink_server` is not set, deserialize into `vl_server`, and then at the end move it over.

Reply:
Yes, when testing this I was able to skip deserialization if `m->varlink_server` exists (for the `systemctl daemon-reload` case).

I noticed in the rest of the deserialization cases we close and replace the file descriptors regardless, so I did this for consistency. Thinking about it again, maybe the other code was written that way for code simplicity. If that's the case then yea I can update it to what you're suggesting.